src/WindowsAzureMessaging/WindowsAzureMessaging/SBNotificationHub.m

```diff
@@ -79,11 +79,14 @@ - (SBNotificationHub*) initWithConnectionString:(NSString*) connectionString not
 
 - (NSString *)convertDeviceToken:(NSData *)deviceTokenData
 {
-    NSString* newDeviceToken = [[[[[deviceTokenData description]
-                                   stringByReplacingOccurrencesOfString:@"<"withString:@""]
-                                  stringByReplacingOccurrencesOfString:@">" withString:@""]
-                                 stringByReplacingOccurrencesOfString: @" " withString: @""] uppercaseString];
-    return newDeviceToken;
+    const char *data = [deviceTokenData bytes];
+    NSMutableString *newDeviceToken = [NSMutableString string];
+    
+    for (NSUInteger i = 0; i < [deviceTokenData length]; i++) {
+        [newDeviceToken appendFormat:@"%02.2hhx", data[i]];
```

Comment:
The previous code produced a token that was all uppercase (via the `uppercaseString` method).

The string format should be `%02.2hhX`.